### PR TITLE
Enhancement: Should generate alarm when VPN is disconnected unexpectedly #22

### PR DESCRIPTION
**Pull request by blackfirespace**

### CODE REVIEW

**Comment by MelvinTo** [comment]
* need provide notification description in en.json for two types of alarms.

File: net2/HostManager.js [hunk 1]
```diff
@@ -1408,6 +1428,27 @@ module.exports = class HostManager {
                   }
                 }
               }
+
```
Comment (jasonlyc) [inline net2/HostManager.js]: VPN Connect alarm should be generated only if the connection is successfully reestablished.

File: net2/HostManager.js [hunk 2]
```diff
@@ -1411,6 +1431,31 @@ module.exports = class HostManager {
...
+                      'p.vpn.strictvpn': settings && settings.strictVPN || false
+                    }
+                  );
+                  await alarmManager2.enqueueAlarm(alarm);
+                }
+              }
+              if (updatedPolicy) {
+                updatedPolicy.waitresume = 0;
+                updatedPolicy.running = true;
+                await this.setPolicyAsync("vpnClient", updatedPolicy);
```
Comment (jasonlyc) [inline net2/HostManager.js]: do not call setPolicy in the synchronized part of this function. Return the updated part instead.

File: net2/HostManager.js [hunk 3]
```diff
@@ -1446,7 +1480,43 @@ module.exports = class HostManager {
...
+                          'p.vpn.subtype': settings && settings.subtype,
+                          'p.vpn.devicecount': device_cout,
+                          'p.vpn.displayname': (settings && (settings.displayName || settings.serverBoxName)) || profileId,
+                          'p.vpn.time': broken_time,
+                          'p.vpn.strictvpn': settings && settings.strictVPN || false
+                        }
+                      );
+                      await alarmManager2.enqueueAlarm(alarm);
+                      updatedPolicy.waitresume = (settings.strictVPN) ? 1 : 0;
+                      await this.setPolicyAsync("vpnClient", updatedPolicy);
```
Comment (jasonlyc) [inline net2/HostManager.js]: It's okay to call setPolicy here since it is in the async part of this function.

File: net2/HostManager.js [hunk 4]
```diff
@@ -1485,6 +1555,11 @@ module.exports = class HostManager {
                 await vpnClientEnforcer.unenforceInterfaceDNSRedirect(intfName, ovpnClient.getInterfaceName(), dnsServers);
               }
             }
+
+            const updatedPolicy = this.policy["vpnClient"];
+            if (!updatedPolicy) return;
+            updatedPolicy.waitresume = 0;
+            await this.setPolicyAsync("vpnClient", updatedPolicy);
```
Comment (jasonlyc) [inline net2/HostManager.js]: Same here. Return the updated part in the return variable instead

File: net2/HostManager.js (hunk 2 above)
Comment (jasonlyc) [inline net2/HostManager.js]: do not call setPolicy in the synchronized part of this function. Return the updated part instead.

Comment (jasonlyc) [reply net2/HostManager.js]: Otherwise, it may trigger concurrent execution of this function

**Pull request merged**